tarantool/response.py

```diff
@@ -54,11 +54,20 @@ def __init__(self, conn, response):
             # Get rid of the following warning.
             # > PendingDeprecationWarning: encoding is deprecated,
             # > Use raw=False instead.
-            unpacker = msgpack.Unpacker(use_list=True, raw=False)
+            if msgpack.version >= (1, 0, 0):
+                unpacker = msgpack.Unpacker(use_list=True, raw=False, strict_map_key=False)
+            else:
+                unpacker = msgpack.Unpacker(use_list=True, raw=False)
         elif conn.encoding is not None:
-            unpacker = msgpack.Unpacker(use_list=True, encoding=conn.encoding)
+            if msgpack.version >= (1, 0, 0):
+                unpacker = msgpack.Unpacker(use_list=True, encoding=conn.encoding, strict_map_key=False)
+            else:
+                unpacker = msgpack.Unpacker(use_list=True, encoding=conn.encoding)
         else:
-            unpacker = msgpack.Unpacker(use_list=True)
+            if msgpack.version >= (1, 0, 0):
+                unpacker = msgpack.Unpacker(use_list=True, strict_map_key=False)
+            else:
+                unpacker = msgpack.Unpacker(use_list=True)
```

Comment:
We can reduce code duplication by unpacking a dictionary to keyword parameters. A kind of this (not tested):

```python
kwargs = dict(use_list=True)
if msgpack.version >= (1, 0, 0):
    # XXX: Explain why it is necessary.
    kwargs['strict_map_key'] = False
if msgpack.version >= (0, 5, 2) and conn.encoding == 'utf-8':
    # Get rid of the following warning.
    # > PendingDeprecationWarning: encoding is deprecated,
    # > Use raw=False instead.
    kwargs['raw'] = False
elif conn.encoding is not None:
    kwargs['encoding'] = conn.encoding

unpacker = msgpack.Unpacker(**kwargs)
```